### ISSUE TITLE
Cannot reproduce performance in paper

### ISSUE THREAD

**Issue by CarolRen233** [basveeling/keras-gcnn#9]
Hi Bas，

I am very insterested in your work. Now I am trying to reproduce the model in your paper Rotation Equivariant  CNNs for Digital Pathology. 

Here is the model I modifies based on the [densenetnew.py](https://github.com/basveeling/keras-gcnn/blob/master/keras_gcnn/applications/densenetnew.py):

```
__________________________________________________________________________________________________
Layer (type)                    Output Shape         Param #     Connected to                     
==================================================================================================
input_1 (InputLayer)            (None, 96, 96, 3)    0                                            
__________________________________________________________________________________________________
initial_Gconv2D (GConv2D)       (None, 94, 94, 192)  648         input_1[0][0]                    
__________________________________________________________________________________________________
dense_0_0_bn (GBatchNorm)       (None, 94, 94, 192)  96          initial_Gconv2D[0][0]            
__________________________________________________________________________________________________
activation_1 (Activation)       (None, 94, 94, 192)  0           dense_0_0_bn[0][0]               
__________________________________________________________________________________________________
cropping2d_1 (Cropping2D)       (None, 92, 92, 192)  0           initial_Gconv2D[0][0]            
__________________________________________________________________________________________________
dense_0_0_Gconv2D (GConv2D)     (None, 92, 92, 24)   5184        activation_1[0][0]               
__________________________________________________________________________________________________
concatenate_1 (Concatenate)     (None, 92, 92, 216)  0           cropping2d_1[0][0]               
                                                                 dense_0_0_Gconv2D[0][0]          
__________________________________________________________________________________________________
tr_0_bn (GBatchNorm)            (None, 92, 92, 216)  108         concatenate_1[0][0]              
__________________________________________________________________________________________________
activation_2 (Activation)       (None, 92, 92, 216)  0           tr_0_bn[0][0]                    
__________________________________________________________________________________________________
tr_0_Gconv2D (GConv2D)          (None, 92, 92, 216)  5832        activation_2[0][0]               
__________________________________________________________________________________________________
average_pooling2d_1 (AveragePoo (None, 46, 46, 216)  0           tr_0_Gconv2D[0][0]               
__________________________________________________________________________________________________
dense_1_0_bn (GBatchNorm)       (None, 46, 46, 216)  108         average_pooling2d_1[0][0]        
__________________________________________________________________________________________________
activation_3 (Activation)       (None, 46, 46, 216)  0           dense_1_0_bn[0][0]               
__________________________________________________________________________________________________
cropping2d_2 (Cropping2D)       (None, 44, 44, 216)  0           average_pooling2d_1[0][0]        
__________________________________________________________________________________________________
dense_1_0_Gconv2D (GConv2D)     (None, 44, 44, 24)   5832        activation_3[0][0]               
__________________________________________________________________________________________________
concatenate_2 (Concatenate)     (None, 44, 44, 240)  0           cropping2d_2[0][0]               
                                                                 dense_1_0_Gconv2D[0][0]          
__________________________________________________________________________________________________
tr_1_bn (GBatchNorm)            (None, 44, 44, 240)  120         concatenate_2[0][0]              
__________________________________________________________________________________________________
activation_4 (Activation)       (None, 44, 44, 240)  0           tr_1_bn[0][0]                    
__________________________________________________________________________________________________
tr_1_Gconv2D (GConv2D)          (None, 44, 44, 240)  7200        activation_4[0][0]               
__________________________________________________________________________________________________
average_pooling2d_2 (AveragePoo (None, 22, 22, 240)  0           tr_1_Gconv2D[0][0]               
__________________________________________________________________________________________________
dense_2_0_bn (GBatchNorm)       (None, 22, 22, 240)  120         average_pooling2d_2[0][0]        
__________________________________________________________________________________________________
activation_5 (Activation)       (None, 22, 22, 240)  0           dense_2_0_bn[0][0]               
__________________________________________________________________________________________________
cropping2d_3 (Cropping2D)       (None, 20, 20, 240)  0           average_pooling2d_2[0][0]        
__________________________________________________________________________________________________
dense_2_0_Gconv2D (GConv2D)     (None, 20, 20, 24)   6480        activation_5[0][0]               
__________________________________________________________________________________________________
concatenate_3 (Concatenate)     (None, 20, 20, 264)  0           cropping2d_3[0][0]               
                                                                 dense_2_0_Gconv2D[0][0]          
__________________________________________________________________________________________________
tr_2_bn (GBatchNorm)            (None, 20, 20, 264)  132         concatenate_3[0][0]              
__________________________________________________________________________________________________
activation_6 (Activation)       (None, 20, 20, 264)  0           tr_2_bn[0][0]                    
__________________________________________________________________________________________________
tr_2_Gconv2D (GConv2D)          (None, 20, 20, 264)  8712        activation_6[0][0]               
__________________________________________________________________________________________________
average_pooling2d_3 (AveragePoo (None, 10, 10, 264)  0           tr_2_Gconv2D[0][0]               
__________________________________________________________________________________________________
dense_3_0_bn (GBatchNorm)       (None, 10, 10, 264)  132         average_pooling2d_3[0][0]        
__________________________________________________________________________________________________
activation_7 (Activation)       (None, 10, 10, 264)  0           dense_3_0_bn[0][0]               
__________________________________________________________________________________________________
cropping2d_4 (Cropping2D)       (None, 8, 8, 264)    0           average_pooling2d_3[0][0]        
__________________________________________________________________________________________________
dense_3_0_Gconv2D (GConv2D)     (None, 8, 8, 24)     7128        activation_7[0][0]               
__________________________________________________________________________________________________
concatenate_4 (Concatenate)     (None, 8, 8, 288)    0           cropping2d_4[0][0]               
                                                                 dense_3_0_Gconv2D[0][0]          
__________________________________________________________________________________________________
tr_3_bn (GBatchNorm)            (None, 8, 8, 288)    144         concatenate_4[0][0]              
__________________________________________________________________________________________________
activation_8 (Activation)       (None, 8, 8, 288)    0           tr_3_bn[0][0]                    
__________________________________________________________________________________________________
tr_3_Gconv2D (GConv2D)          (None, 8, 8, 288)    10368       activation_8[0][0]               
__________________________________________________________________________________________________
average_pooling2d_4 (AveragePoo (None, 4, 4, 288)    0           tr_3_Gconv2D[0][0]               
__________________________________________________________________________________________________
dense_4_0_bn (GBatchNorm)       (None, 4, 4, 288)    144         average_pooling2d_4[0][0]        
__________________________________________________________________________________________________
activation_9 (Activation)       (None, 4, 4, 288)    0           dense_4_0_bn[0][0]               
__________________________________________________________________________________________________
cropping2d_5 (Cropping2D)       (None, 2, 2, 288)    0           average_pooling2d_4[0][0]        
__________________________________________________________________________________________________
dense_4_0_Gconv2D (GConv2D)     (None, 2, 2, 24)     7776        activation_9[0][0]               
__________________________________________________________________________________________________
concatenate_5 (Concatenate)     (None, 2, 2, 312)    0           cropping2d_5[0][0]               
                                                                 dense_4_0_Gconv2D[0][0]          
__________________________________________________________________________________________________
final_bn (GBatchNorm)           (None, 2, 2, 312)    156         concatenate_5[0][0]              
__________________________________________________________________________________________________
activation_10 (Activation)      (None, 2, 2, 312)    0           final_bn[0][0]                   
__________________________________________________________________________________________________
group_pool_1 (GroupPool)        (None, 2, 2, 39)     0           activation_10[0][0]              
__________________________________________________________________________________________________
global_average_pooling2d_1 (Glo (None, 39)           0           group_pool_1[0][0]               
__________________________________________________________________________________________________
dense_1 (Dense)                 (None, 1)            40          global_average_pooling2d_1[0][0] 
==================================================================================================
Total params: 66,460
Trainable params: 65,830
Non-trainable params: 630
__________________________________________________________________________________________________

```
I use this above model to train on [pcam](https://github.com/basveeling/pcam)  data. However, I got acc of 0.6 and auc of 0.6. I would be very appreciate if you can help to check what are the problems with this model and is there anything I didn't notice?

Other parameters:
(batch_size=32, conv_group='D4', depth=9, dropout_rate=0.0, epochs=50, growth_rate=3, img_channels=3, nb_classes=1, nb_dense_block=5, nb_filter=24, network='GDenseNet', wh=96, padding = 'valid', include_top = True)

**Comment by MohammedHAlali** [comment]
I'm also interested in reproducing the paper's results.
The paper reported a use of batch_size=64, not 32.

What optimizer and learning rate did you use?
It will be very helpful if you copy/paste your optimizer setting code.

I hope it will work for you.
